### DISCUSSION TITLE
QOL improvement

### DISCUSSION THREAD

**Discussion by BioTox1n** [n4ze3m/page-assist#93] [Ideas]
Is it possible to have two models loaded simultaneously when the GPU has enough VRAM?
When using context from the web pages, the current model gets unloaded for the Embedding model to do its work and then the main model re-loads.
This adds unnecessary wait time.
If possible, having a switch to disable offloading would be nice so that the embedder can work whenever needed without adding extra time.

**Comment by n4ze3m** [comment]
Yes, I am currently adding the supported model configuration from Ollama to the Page Assist settings page. :) #90

**Comment by n4ze3m** [comment]
Hey, model parameter settings have been added in the v1.1.10 release. There are two ways to configure it: one, globally in the Ollama settings, and two, for the current chat by clicking the brain with the cog icon in the header.

More parameter settings will be added in upcoming updates. :)

![image](https://github.com/n4ze3m/page-assist/assets/39720973/a998402e-2c7a-429e-b946-40148b85a907)
![image](https://github.com/n4ze3m/page-assist/assets/39720973/8228c921-54e1-439f-92a2-0c3eb57a15c3)

**Reply by BioTox1n** [reply]
I'm glad to hear about the update, as it addresses some necessary settings. However, my original question was about loading two models simultaneously (e.g., LLAMA3 and an Embedding model). I want to keep my main model and the Embedding model in memory while using the internet search feature.

So far I've noticed that the main model unloads for the Embedding model to do its thing and then re-loads. 
![Screenshot](https://github.com/n4ze3m/page-assist/assets/143258099/870a33ac-6d57-4c2d-a059-3b8b5e863e46)


I found that the relevant flags for this feature are OLLAMA_ NUM_PARALLEL and OLLAMA_MAX_LOADED_MODELS, but I'm unsure of the implementation of them since I'm not a programmer.

**Reply by n4ze3m** [reply]
Hey, this video might be helpful for you in setting up the Ollama environment for loading models in parallel: https://www.youtube.com/watch?v=MDbdb-W4x4w Let me know if you have any other issues :)
